test/specs/01.displayname.spec.js

```diff
@@ -0,0 +1,93 @@
+// Unit Test that are related to annotations should be here
+
+var includes = require(__dirname + '/../utils/ermrest-init.js').init();
+
+var server = includes.server;
+var ermRest = includes.ermRest;
+var ermrestUtils = includes.ermrestUtils;
+
+describe('Display Name: ', function () {
+    var catalog_id, schemaName, schema, catalog;
+
+    // This function should be present in all spec files. It will add sample database and configurations.
+    beforeAll(function (done) {
+        ermrestUtils.importData({
+            setup: require('../configuration/displayname.spec.conf.json'),
+            url: includes.url ,
+            authCookie : includes.authCookie
+        }).then(function(data) {
+            catalog_id = data.catalogId;
+            schemaName = data.schema.name;
+            console.log("Data imported with catalogId " + data.catalogId);
+            done();
+        }, function(err) {
+            console.log("Unable to import data");
+            console.dir(err);
+            done.fail();
+        });
+    });
+
+    // Test Cases:
+
+    it('schema should use its own heuristics.', function(done) {
```

Comment:
Yes, these are the right level of specificity, as far as I’m concerned. thanks

From: Aref Shafaei <notifications@github.com>
Reply-To: informatics-isi-edu/ermrestjs <reply@reply.github.com>
Date: Wednesday, July 13, 2016 at 2:06 PM
To: informatics-isi-edu/ermrestjs <ermrestjs@noreply.github.com>
Cc: Robert Schuler <schuler@isi.edu>, Mention <mention@noreply.github.com>
Subject: Re: [informatics-isi-edu/ermrestjs] Displayname Unit Tests (#82)


In test/specs/01.displayname.spec.js<https://github.com/informatics-isi-edu/ermrestjs/pull/82#discussion_r70708753>:

> +            authCookie : includes.authCookie

> +        }).then(function(data) {

> +            catalog_id = data.catalogId;

> +            schemaName = data.schema.name;

> +            console.log("Data imported with catalogId " + data.catalogId);

> +            done();

> +        }, function(err) {

> +            console.log("Unable to import data");

> +            console.dir(err);

> +            done.fail();

> +        });

> +    });

> +

> +    // Test Cases:

> +

> +    it('schema should use its own heuristics.', function(done) {

@robes<https://github.com/robes> I know that the descriptions that I used are too generic. But I don't want them to be very specific to the data as well.
I changed them and the success output looks like this for now:

  For determining display name,

    ✓ schema should use name styles (`underline_space`, `title_case`) that are defined in its display annotations.

    ✓ table should use its own name that is defined in display annotation.

    ✓ table should use its own name styles (`underline_space`, `title_case`) that are defined in its display annotation.

    ✓ table should use schema name styles when `name_style` and `name` are not defined in its display annotation.

    ✓ column should use its own name that is defined in display annotation.

    ✓ column should use its own name styles (`underline_space`, `title_case`) that are defined in its display annotation.

    ✓ column should use table name styles when `name_style` and `name` are not defined in its display annotation.

    ✓ column should use schema name styles when `name_style` and `name` are not defined in its display annotation nor table's display annotation.

Are these good enough?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/informatics-isi-edu/ermrestjs/pull/82/files/c62508c3a60587317c1942206077d66e60dbe07f#r70708753>, or mute the thread<https://github.com/notifications/unsubscribe/AEAhWmXhtcYkD9X84pPU8wmXG_6XCq17ks5qVVNMgaJpZM4JK9IH>.